### DISCUSSION TITLE
Randomize existing tag colors

### DISCUSSION THREAD

**Discussion by jovandeginste** [jonaswinkler/paperless-ng#709] [General]
When I started using Paperless, I auto-created a lot of tags based on folders where my existing documents were living in. All those tags had the same color. Now, there is an almost endless choice for colors, and new tags get random colors suggested. However, I still have those starting-colors for my 150+ tags.

This little Postgres code randomizes the colors in a consistent way:

```sql
update documents_tag set color = concat('#', substring(md5(name), 1, 6));
```

Basically, take the first 6 characters of the md5-sum of the name, which is hexadecimal and thus maps to a color. Uniqueness is not guaranteed, but not necessary either.

Okay, I lied: this is not randomized 😉

**Comment by jonaswinkler** [comment]
Brilliant solution. So simple.

Seems like it was a good idea to get a color picker in there!

**Reply by jovandeginste** [reply]
I was thinking about suggesting it (more color variety, and this for default) for a while. I was also waiting to see where #84 went (but forgot to subscribe to it, apparently)

It's an old trick I took from this project: [Redmine tags plugin](https://github.com/ixti/redmine_tags/blob/1c2e829485c1cb029147b123a93bcee5b9bc81e0/app/helpers/tags_helper.rb#L46)

But it was such a small thing in the grand scheme of things...

**Reply by jovandeginste** [reply]
Also, thanks for the praise (but i can't take credit)
